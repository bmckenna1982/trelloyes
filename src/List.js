import React from 'react';
import Card from './Card';
import './List.css';

function List(props) {
    // const cardArray = props.cards;
        
    return (
        <section className="List">
            <header className="List-header">                
                <h2>{props.header}</h2>
            </header>
            <div className="List-cards">
                {props.cards.map(card => (
                    <Card 
                        key={card.id}
                        title={card.title}
                        content={card.content}
                    />
                ))}
            </div>
            <button type="button" class="List-add-button">
              + Add Random Card
            </button>
        </section>
        ); 
}

export default List;