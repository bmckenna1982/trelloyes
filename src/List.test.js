import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

const cards = [{
    id: 1,
    title: 'test',
    content: 'Testing'
}]

it ('renders without crashing', () => {
    const div = document.createElement('div');    
    ReactDOM.render(<List cards={cards}/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it ('renders the ui as expected', () => {
    const tree = renderer
        .create(<List cards={cards}/>)
        .toJSON();
    expect (tree).toMatchSnapshot();
})