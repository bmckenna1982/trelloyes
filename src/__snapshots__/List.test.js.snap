// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ui as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
                  
      <h3>
        test
      </h3>
                  
      <p>
        Testing
      </p>
    </div>
  </div>
  <button
    className="List-add-button"
    onClick={[Function]}
    type="button"
  >
    + Add Random Card
  </button>
</section>
`;
